lib/opencensus/tags/formatters/binary.rb

```diff
@@ -0,0 +1,142 @@
+# frozen_string_literal: true
+
+module OpenCensus
+  module Tags
+    module Formatters
+      ##
+      # This formatter serializes and deserializes tags context according to
+      # the OpenCensus' BinaryEncoding specification. See
+      # [documentation](https://github.com/census-instrumentation/opencensus-specs/blob/master/encodings/BinaryEncoding.md).
+      #
+      # @example Serialize
+      #
+      # formatter = OpenCensus::Tags::Formatters::Binary.new
+      #
+      # tag_map = OpenCensus::Tags::TagMap.new({\"key1" => \"val1"})
+      # binary = formatter.serialize tag_map # "\x00\x00\x04key1\x04val1"
+      #
+      # @example Deserialize
+      #
+      # formatter = OpenCensus::Tags::Formatters::Binary.new
+      #
+      # binary = "\x00\x00\x04key1\x04val1"
+      # tag_map = formatter.deserialize binary
+      #
+      class Binary
+        # @private
```

Comment:
This exception can be thrown from public methods, so it cannot be `@private`.

Reply:
Fixed